lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetChainId.kt

```diff
@@ -1,20 +1,31 @@
 package com.swmansion.starknet.data.types
 
-import kotlinx.serialization.SerialName
+import com.swmansion.starknet.data.serializers.StarknetChainIdSerializer
 import kotlinx.serialization.Serializable
 
-@Serializable
-enum class StarknetChainId(val value: Felt) {
-    @SerialName("0x534e5f4d41494e")
-    MAINNET(Felt.fromHex("0x534e5f4d41494e")), // encodeShortString('SN_MAIN'),
+@Serializable(with = StarknetChainIdSerializer::class)
+data class StarknetChainId(val value: Felt) {
+    companion object {
+        @field:JvmField
+        val MAIN = StarknetChainId(Felt.fromHex("0x534e5f4d41494e")) // encodeShortString('SN_MAIN'),
 
-    @Deprecated("Consider using SEPOLIA instead")
-    @SerialName("0x534e5f474f45524c49")
-    GOERLI(Felt.fromHex("0x534e5f474f45524c49")), // encodeShortString('SN_GOERLI'),
+        @field:JvmField
+        val GOERLI = StarknetChainId(Felt.fromHex("0x534e5f474f45524c49")) // encodeShortString('SN_GOERLI'),
 
-    @SerialName("0x534e5f5345504f4c4941")
-    SEPOLIA(Felt.fromHex("0x534e5f5345504f4c4941")), // encodeShortString('SN_SEPOLIA'),]
+        @field:JvmField
+        val SEPOLIA = StarknetChainId(Felt.fromHex("0x534e5f5345504f4c4941")) // encodeShortString('SN_SEPOLIA'),
 
-    @SerialName("0x534e5f494e544547524154494f4e5f5345504f4c4941")
-    INTEGRATION_SEPOLIA(Felt.fromHex("0x534e5f494e544547524154494f4e5f5345504f4c4941")), // encodeShortString('SN_INTEGRATION_SEPOLIA'),
+        @field:JvmField
+        val INTEGRATION_SEPOLIA = StarknetChainId(Felt.fromHex("0x534e5f494e544547524154494f4e5f5345504f4c4941")) // encodeShortString('SN_INTEGRATION_SEPOLIA'),
+
+        @JvmStatic
+        fun fromNetworkName(shortString: String): StarknetChainId {
+            return StarknetChainId(Felt.fromShortString(shortString))
+        }
+
+        @JvmStatic
+        fun fromHex(hex: String): StarknetChainId {
+            return StarknetChainId(Felt.fromHex(hex))
+        }
```

Comment:
Should we keep those? Hard for me to definitively say whether those are useful or redundant.

Reply:
I think they may come in handy, I'd leave those

Reply:
Added just one more (`toNetworkName`) for consistency